internal/cmddiff/cmddiff.go

```diff
@@ -92,7 +90,11 @@ func (r *Runner) preRunE(c *cobra.Command, args []string) error {
 		r.DiffType = diff.DiffType(r.diffType)
 	}
 
-	r.Path = dir
+	p, err := pkg.New(dir)
+	if err != nil {
+		return err
+	}
+	r.Path = string(p.DisplayPath)
```

Comment:
What's the semantic of `Path`?

```
type Command struct {
	// Path to the local package directory
	Path string
```

The spirit of using `Package` library is to avoid passing raw strings around which have ambiguous semantics.

Furthermore, you should only use `DisplayPath` for display purposes. Diff command uses this path to do io operations.

```
// Slash-separated path to the package directory on the filesytem relative to current working directory.
// This is not guaranteed to be unique (e.g. in presence of symlinks) and should only
// be used for display purposes and is subject to change.
```

Reply:
Updated.

I will address the standardizing part in coming PRs.

Reply:
To make sure we're on the same page: Path field in Command struct should not be of type string. When dealing with package, pass a Package. Otherwise, pass UniquePath.